Source/Core/DolphinQt2/Config/FilesystemWidget.cpp

```diff
@@ -4,6 +4,290 @@
 
 #include "DolphinQt2/Config/FilesystemWidget.h"
 
-FilesystemWidget::FilesystemWidget(const GameFile& game) : m_game(game)
+#include <QCoreApplication>
+#include <QFileDialog>
+#include <QHeaderView>
+#include <QMenu>
+#include <QMessageBox>
+#include <QProgressDialog>
+#include <QStandardItemModel>
+#include <QTreeView>
+#include <QVBoxLayout>
+
+#include <future>
+
+#include "DiscIO/DiscExtractor.h"
+#include "DiscIO/Enums.h"
+#include "DiscIO/Filesystem.h"
+#include "DolphinQt2/Resources.h"
+
+constexpr int ENTRY_PARTITION = Qt::UserRole;
+constexpr int ENTRY_NAME = Qt::UserRole + 1;
+constexpr int ENTRY_TYPE = Qt::UserRole + 2;
+
+enum class EntryType
+{
+  TYPE_DISC = -2,
```

Comment:
Since this is an enum class, the `TYPE_` can be dropped and the name should be cased like `Disc`, etc.